### ISSUE TITLE
Counting intronic reads in single-cell RNA-seq data using Alevin

### ISSUE THREAD

**Issue by vertesy** [COMBINE-lab/salmon#450]
### Large intronic read proportion in our latest single-cell RNA-seq  data set

We realised that around 35% of our mapped reads in our recent single-cell RNA-seq experiment because map to introns.

<img width="400" alt="image" src="https://user-images.githubusercontent.com/5101911/69015737-65272f80-0997-11ea-87ab-9237e105622a.png">

### Loss of reads by using exonic references

Both Alevin and Cell ranger count only reads aligned to exons by default, thus losing some 35% of the reads. Large intronic proportions are a general feature of single-cell RNA-seq, for example the 10X PBMCs has ~20-25%, and nuclear data sets have 46% intronic reads in [[source](https://kb.10xgenomics.com/hc/en-us/articles/360000087552-Why-do-I-have-a-high-percentage-of-reads-mapping-to-intronic-regions-)]. 

**So the best way to increase library 'quality'  / counting statistics (which is my prime concern) seem to be including reads mapped to introns – leading to hopefully a large increase in UMIs.**

### Considerations: separate, additive signals

Since exons and introns contain different biological signal, ideally I would like to count as:

1. Exon only 		(current)* 
2. Intron only 		(as control / comparison to 1.) 
3. Exon + intron 	(finally, for boosting statistics) 

Where 2 would be a complementary set to 1, so that no read is counted as both intron and exon, and thus 1+2  =~ 3. 

\* _I am not sure what happens to a read that maps to an exon/intron junction using the default Alevin and 10x Cell Ranger pipelines, but I assume both count it as exonic._

### Solution by counting programs

- There is a way to build a "pre-mRNA" reference for 10X Cell ranger as described [here](https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/advanced/references#premrna). 

- I wondered if Alevin can be adapted to this aim? I guess besides all complete transcript models, you need to add one more: the pre-mRNA. (While some complexity can theoretically arise by the serial nature of splicing, I would ignore it at first)

Would that be a good idea, or the advantage of Alevin would evaporate if we include introns?

Thanks!

**Comment by k3yavi** [comment]
Hi @vertesy ,

Thanks for asking the very interesting question.
I'd say the answer might depend on what's your downstream use case. Traditionally, no quantification pipeline has used the pre-mRNA counts alone to bump up the gene counts, however, recent method of estimating RNA-velocity does utilizes the intronic counts for extracting the ratio of spliced/unspliced counts.

If you are interested in disjoint signals (gene count matrix) for spliced and unspliced molecules you can use the recent scheme of decoy indexing from our latest [preprint](https://www.biorxiv.org/content/10.1101/657874v2). We (mostly @csoneson) have been testing alevin with following scheme for generating spliced and unspliced counts.

1.) Spliced Counts: Index transcriptome w/ pre-mRNA sequence as the decoys.
2) Unspliced Counts: Index pre-mRNA sequence w/ transcriptome as the decoys.

The third case is a little tricky because if you index both pre-mRNA and transcriptome, due to relatively longer length of pre-mRNA sequence compared to transcripts it might end-up biasing the UMI deduplication algorithm towards unspliced counts.

To summarize, the best way to have an additive spliced and unspliced counts is still an open area of research.

**Comment by mdshw5** [comment]
@k3yavi It seems like indexing both mature and un-spliced transcripts in one index, quantifying them jointly, and then post processing (re-normalizing) the two feature types could be a bit more efficient. Is there any obvious advantage to explicitly specifying which features are decoys vs. mapping and quantifying everything in one go?

**Comment by vertesy** [comment]
Thanks @k3yavi, I will look into it.

**Comment by stephaniehicks** [comment]
Thanks @vertesy for your question and @k3yavi for your response! I have a question about @k3yavi 's comment: 
 
 > The third case is a little tricky because if you index both pre-mRNA and transcriptome, due to relatively longer length of pre-mRNA sequence compared to transcripts it might end-up biasing the UMI deduplication algorithm towards unspliced counts.

If you have single-nucleus RNA-seq data where theoretically the reads should be coming from the pre-mRNA transcripts (but still could have reads mapping to mRNA transcripts), would it make sense to include both the pre-mRNA and mRNA transcripts in the index with a goal biasing the UMI dedup algorithm towards unspliced counts? 

Also, on an unrelated note, would this be possible with either Alevin or Salmon? Thank you!

**Comment by k3yavi** [comment]
My apologies for the late reply. Please checkout our latest preprint [here](https://www.biorxiv.org/content/10.1101/2020.03.13.990069v1.full.pdf).
We explore the intricacies with generating unspliced counts in detail .

**Comment by stephaniehicks** [comment]
Thanks @k3yavi. I'll check out the preprint.

**Comment by k3yavi** [comment]
Please check the following tutorial for ste-by-step process
https://combine-lab.github.io/alevin-tutorial/2020/alevin-velocity/
